src/realm/array_flex.cpp

```diff
@@ -177,13 +177,13 @@ bool ArrayFlex::find_all(const Array& arr, int64_t value, size_t start, size_t e
 
 template <typename Cond, bool v>
 inline size_t ArrayFlex::parallel_subword_find(const Array& arr, uint64_t value, size_t width_mask, size_t offset,
-                                               uint_least8_t width, size_t start, size_t end) const
+                                               size_t width, size_t start, size_t end) const
 {
-    const auto MSBs = populate(width, width_mask);
-    const auto search_vector = populate(width, value);
-    const auto field_count = num_fields_for_width(width);
-    const auto bit_count_pr_iteration = num_bits_for_width(width);
-    auto total_bit_count_left = static_cast<signed>(end - start) * width;
+    const auto MSBs = populate(static_cast<int>(width), width_mask);
+    const auto search_vector = populate(static_cast<int>(width), value);
+    const auto field_count = num_fields_for_width(static_cast<int>(width));
+    const auto bit_count_pr_iteration = num_bits_for_width(static_cast<int>(width));
+    int64_t total_bit_count_left = static_cast<int64_t>((end - start) * width);
```

Comment:
it would also be ok to use size_t for this ... perhaps that would allow for dropping a cast below?

Reply:
I think `total_bit_count_left` can become negative since we are taking out the amount of bits read per iteration. I'll address all the casts in a separate PR.. let's merge https://github.com/realm/realm-core/pull/7427